packages/office-ui-fabric-react/src/components/pickers/AutoFill/BaseAutoFill.tsx

```diff
@@ -67,15 +67,18 @@ export class BaseAutoFill extends BaseComponent<IBaseAutoFillProps, IBaseAutoFil
   }
 
   public componentWillReceiveProps(nextProps: IBaseAutoFillProps) {
+    let newValue;
+
     if (this.props.updateValueInWillReceiveProps) {
-      let newValue = this.props.updateValueInWillReceiveProps();
+      newValue = this.props.updateValueInWillReceiveProps();
+    }
 
-      if (newValue !== null) {
-        this._value = newValue;
-      }
+    if (this._autoFillEnabled && this._doesTextStartWith(nextProps.suggestedDisplayValue!, newValue ? newValue : this._value)) {
+      newValue = nextProps.suggestedDisplayValue;
     }
-    if (this._autoFillEnabled && this._doesTextStartWith(nextProps.suggestedDisplayValue!, this._value)) {
-      this.setState({ displayValue: nextProps.suggestedDisplayValue });
+
+    if (!!newValue) {
```

Comment:
You might want to check for undefined here, if the new value is "" then !!newValue would be false and it would skip the state update